### ISSUE TITLE
TypeError: 'int' object is not iterable`

### ISSUE THREAD

**Issue by docsuleman** [ABI-Software-Laboratory/ICMA#5]
I am using python 3.6 and managed to install ICMA by making some changes to code to make it compatible with with python 3.6. When I press track this is the error I get. 

`Traceback (most recent call last):
  File "C:\Users\intag\Downloads\Compressed\ICMA-master\ICMA-master\Python\ICMA2.0\speckleTracking\TrackAndView.py", line 997, in startTracking
    self.speckleTracker.run()
  File "C:\Users\intag\Downloads\Compressed\ICMA-master\ICMA-master\Python\ICMA2.0\speckleTracking\TrackAndView.py", line 579, in run
    self.allFramesTracking()
  File "C:\Users\intag\Downloads\Compressed\ICMA-master\ICMA-master\Python\ICMA2.0\speckleTracking\TrackAndView.py", line 597, in allFramesTracking
    self.capHeart.setupBasePlaneVariation(endSystoleFrame, [self.capHeart.basePlanePoints,basePlanePoints])
  File "C:\Users\intag\Downloads\Compressed\ICMA-master\ICMA-master\Python\ICMA2.0\capmodel\LV.py", line 304, in setupBasePlaneVariation
    normal, position = self.interpolatePlanes(sysTime,basePlanes, 0)
  File "C:\Users\intag\Downloads\Compressed\ICMA-master\ICMA-master\Python\ICMA2.0\capmodel\LV.py", line 188, in interpolatePlanes
    dist = np.fabs(ptimes-time)
TypeError: 'int' object is not iterable`

this error corresponds to following function
`def interpolatePlanes(self,systime,planeParameters,time):
       
        ptimes = np.array([0.0,systime])
        #If the time is close to a specified base plane, return that one
      

        dist = np.fabs(ptimes-time)
        ix = np.argsort(dist)[0]
        if dist[ix] < 1e-6:
            return planeParameters[ix]
        #Handle edge cases
        if time <= ptimes[0]:
            return planeParameters[0]
        elif time > ptimes[-1]: #This could be end systole frame, so provide the first frame that corresponds to end-diastole
            startFrame = planeParameters[-1]
            startTime = ptimes[-1]
            endFrame   = planeParameters[0]
            endTime = self.times[-1]
        else:
            startFrame = planeParameters[0]
            startTime = ptimes[0]
            endFrame   = planeParameters[-1]
            endTime = self.times[-1]            
        weight = (float(time)-float(startTime))/(float(endTime)-float(startTime))
        normal = startFrame[0] + weight*(endFrame[0]-startFrame[0])
        position = startFrame[1] + weight*(endFrame[1]-startFrame[1])
        return [normal,position]
`

I'm a newbie at python so sorry if its a stupid question.

**Comment by Jagirhussan** [comment]
Could you print and let me know the ouput of
systime, ptimes and dist 
before the line 
ix = np.argsort(dist)[0]?

**Comment by docsuleman** [comment]
thanks, @Jagirhussan  for your reply. I was able to fix the issue by changing         

`endSystoleFrame = np.min(self.baseplaneLocations.keys())`
 to
`endSystoleFrame = np.min(list(self.baseplaneLocations.keys()))`

on TrackAndView.py#L714

I also needed to change many similar lines too to make this code working. After getting back to my hospital,  I'll test on real patients soon and will see how this turns out. for now, I have few compressed DICOM files where it's not tracking the border properly.
